### PR TITLE
init socket provider

### PR DESCRIPTION
**Pull request by Sparksx**
Can create connection with an indoor location emitter
Can add and remove listeners on this connection

See README.md file for more details

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,53 @@
...
+- run `cd /path/to/project/socket-indoor-location-provider-js`
+- run `webpack` if you only want to build it
+- run `webpack --progress --colors --watch` to rebuild on each file update
+
+
+- Import `dist/app.bundle.js` in you project:   
+`<script charset="utf-8" type="text/javascript" src="dist/app.bundle.js"></script>`
+- Create new instance of `MwzSocketIndoorLocationProvider` class
+````javascript
+var indoorLocationInstance = new MwzSocketIndoorLocationProvider('http://localhost:3003', '127.0.0.1');
```
Comment (mathieugerard) [inline README.md]: socketIndoorLocationProvider =

File: README.md [hunk 2]
```diff
@@ -1,2 +1,53 @@
...
+// Available params:
+// serverUrl: Socket emitter url
+// userId: String to identifie an user
+// errorCallback: function called in case of error
```
Comment (mathieugerard) [inline README.md]: We need to discuss the error callback

File: package.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  "name": "socket-indoor-location-provider",
+  "version": "1.0.0",
+  "description": "IndoorLocationProvider using websockets to retrieve the user position from a server",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/Sparksx/socket-indoor-location-provider-js.git"
```
Comment (mathieugerard) [inline package.json]: neets to be github.com/IndoorLocation

File: README.md [hunk 4]
```diff
@@ -1,2 +1,53 @@
 # socket-indoor-location-provider-js
 IndoorLocationProvider using websockets to retrieve the user position from a server
+
```
Comment (mathieugerard) [inline README.md]: We need to add somewhere that the code is in Typescript

File: package.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/Sparksx/socket-indoor-location-provider-js/issues"
+  },
+  "homepage": "https://github.com/Sparksx/socket-indoor-location-provider-js#readme",
+  "dependencies": {
+    "socket.io-client": "2.0.3"
+  },
+  "devDependencies": {
+    "awesome-typescript-loader": "^3.2.3",
```
Comment (mathieugerard) [inline package.json]: I prefer to fix package versions

File: src/index.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    timestamp: number;
+    accuracy: number;
+}
+
+interface UserIndoorLocation {
+    userId: string;
+    indoorLocation: IndoorLocation;
+}
+
+export class MwzSocketIndoorLocationProvider {
```
Comment (mathieugerard) [inline src/index.ts]: why MWZ ?

File: src/index.ts [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    protected connection: any;
+    protected listeners: {(p: UserIndoorLocation): void}[];
+
+    constructor(serverUrl: string, userId: string, errorCallback: {(err: any):void}) {
```
Comment (mathieugerard) [inline src/index.ts]: serverUrl or socketUrl ?

File: README.md [hunk 8]
```diff
@@ -1,2 +1,53 @@
...
+````javascript
+indoorLocationInstance.start();
+indoorLocationInstance.stop();
+
+indoorLocationInstance.isStarted(); // return boolean
+````
+
+`MwzSocketIndoorLocationProvider` works with listeners, you've to subscribe to events:
+````javascript
+var listener = function (indoorLocation) {
```
Comment (mathieugerard) [inline README.md]: Need to document the object retrieved in the callback. userIndoorLocation

File: src/index.ts [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+        self.listeners = [];
+        self.connection = io(serverUrl, {
+            autoConnect: false,
+            query: {
+                userId: userId
+            }
+        });
+
+        this.connection.on('connect', function () {
+            console.log('connection started');
```
Comment (mathieugerard) [inline src/index.ts]: Would be awesome to have a onStart callback

File: src/index.ts [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+    public getName() {
+        return 'socket';
+    }
+}
+
+// Needed ton expose MwzSocketIndoorLocationProvider as global
+declare global {
+    interface Window { MwzSocketIndoorLocationProvider: any; }
+}
+window.MwzSocketIndoorLocationProvider = MwzSocketIndoorLocationProvider || {};
```
Comment (mathieugerard) [inline src/index.ts]: This makes it not usable in node.js. Any other option?

File: README.md (hunk 8 above)
Comment (Sparksx) [reply README.md]: Done

File: README.md (hunk 1 above)
Comment (Sparksx) [reply README.md]: done

File: package.json (hunk 3 above)
Comment (Sparksx) [reply package.json]: Done

File: package.json (hunk 5 above)
Comment (Sparksx) [reply package.json]: Done

File: src/index.ts (hunk 6 above)
Comment (Sparksx) [reply src/index.ts]: Removed

File: src/index.ts (hunk 10 above)
Comment (Sparksx) [reply src/index.ts]: Fixed

File: src/index.ts (hunk 9 above)
Comment (Sparksx) [reply src/index.ts]: I have two solutions, we need to discuss about it

File: README.md (hunk 4 above)
Comment (Sparksx) [reply README.md]: done

**Pull request merged**